content/en/service_management/app_builder/expressions.md

```diff
@@ -0,0 +1,121 @@
+---
+title: JavaScript Expressions
+kind: documentation
+further_reading:
+- link: "/service_management/app_builder/build/"
+  tag: "Documentation"
+  text: "Build Apps"
+- link: "/service_management/app_builder/components/"
+  tag: "Documentation"
+  text: "Components"
+---
+
+You can use JavaScript (JS) expressions anywhere in App Builder to create custom interactions between the different parts of your app. As you type an expression, App Builder offers autocomplete suggestions based on the existing queries and components in your app. Click on an autocomplete suggestion to use it in your expression, or use the arrow keys on your keyboard and make a selection with the Enter key.
+
+Some fields, like the [post-query transformation][1], display a code editor by default and accept plain JS. In all other fields, enclose your JS expressions in `${}`. For example, to interpolate the values of two text input components named `textInput0` and `textInput1` into the **Content** property of a text component (and add an exclamation mark), use the expression `${textInput0.value} ${textInput1.value}!`.
+
+{{< img src="service_management/app_builder/interpolation.png" alt="The text component fills with the words 'Hello' and 'World', each interpolated from a text input component value" style="width:100%;" >}}
+
+App Builder accepts standard vanilla JavaScript syntax, however:
+- The result of the expression must match the result expected by the component or query property. For example, the text component's **Is Visible** property expects a Boolean. To find out what type of data a component property expects, see [View component properties](#view-component-properties).
+- Your code has read-only access to the app state, but App Builder executes the code in a sandboxed environment with no access to the Document Object Model (DOM) or browser APIs.
+
```

Comment:
I see that we've dropped the section on the `?.` operator, but won't it be confusing to users when that shows up in autocompleted strings but we never address or explain it?

Reply:
This is one of those hills.